hive/src/test/java/org/apache/iceberg/hive/TestHiveTableConcurrency.java

```diff
@@ -96,13 +96,14 @@ public synchronized void testConcurrentConnections() throws InterruptedException
         .build();
 
     ExecutorService executorService = MoreExecutors.getExitingExecutorService(
-        (ThreadPoolExecutor) Executors.newFixedThreadPool(10));
+        (ThreadPoolExecutor) Executors.newFixedThreadPool(7));
 
-    for (int i = 0; i < 10; i++) {
+    for (int i = 0; i < 7; i++) {
       executorService.submit(() -> icebergTable.newAppend().appendFile(file).commit());
     }
 
     executorService.shutdown();
-    Assert.assertTrue("Timeout", executorService.awaitTermination(1, TimeUnit.MINUTES));
+    Assert.assertTrue("Timeout", executorService.awaitTermination(2, TimeUnit.MINUTES));
```

Comment:
This test leads to a deadlock in HMS that it recovers from automatically, so we cannot really predict the needed time. Before the fix, we didn't have retries. Therefore, I reduced the number of concurrent connections but it is still higher than the number threads we use for test HMS.